### DISCUSSION TITLE
apt-offline errors on ubuntu server 20.04, upgrade.sig is zero bytes

### DISCUSSION THREAD

**Discussion by buxbaumbixbybrae** [rickysarraf/apt-offline#185] [Q&A] [answered]
Hi,

We wish to fully update / upgrade (and subsequently install new packages on) an offline install of ubuntu server 20.04 which is a fresh install from iso 20.04.4. We are stuck at generating the upgrade.sig file which is zero bytes long.
The online machine is linux mint 20 tricia (ubuntu 18.04 iirc).

We have followed the workflow summarised below and errors were encountered.
In particular when `apt-offline set upgrade.sig --upgrade` is run there are no errors but the upgrade.sig file is 0 bytes which is not expected. Obviously the online machine can't download anything against that.
I'm expecting non-zero output for upgrade.sig and hundreds of MB downloaded when I run get upgrade.sig on the online machine
Any issues with the workflow listed below?
What is working, and what is failed?
The attached file contains more complete output. 
Which errors are of concern and which are not important?
The version of apt-offline on the offline machine is 1.8.1 compared to 1.8.2.1 on the offline machine, could this be an issue?

I'm not sure what I've done wrong. Any tips to get this working are appreciated.

```
online$ download apt-offline_1.8.2-1_all.deb
online$ download python3-magic_0.4.15-3_all.deb
online$ sudo apt-get install apt-offline
offline$ sudo dpkg -i ./python3-magic_0.4.15-3_all.deb
offline$ sudo dpkg -i ./apt-offline_1.8.2-1_all.deb
offline$ sudo apt-offline set update.sig --update
online$ sudo apt-offline get update.sig --threads 1 --bundle update-bundle.zip
## about 23MB downloaded, lots of warnings here, see attached.
offline$ sudo apt-offline install update-bundle.zip
## keyring error but then good signature messages and things are synced.
offline$ sudo apt update
## I believe we should not get web access errors below with everything cached by apt-offline?
All packages are up to date.
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/focal/InRelease  Temporary failure resolving 'archive.ubuntu.com'
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/focal-updates/InRelease  Temporary failure resolving 'archive.ubuntu.com'
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/focal-backports/InRelease  Temporary failure resolving 'archive.ubuntu.com'
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/focal-security/InRelease  Temporary failure resolving 'archive.ubuntu.com'
W: Some index files failed to download. They have been ignored, or old ones used instead.

offline$ sudo apt-offline set upgrade.sig --upgrade --upgrade-type dist-upgrade
## the upgrade.sig output file created is 0 bytes long
```
[apt-offline-diags.txt](https://github.com/rickysarraf/apt-offline/files/8319827/apt-offline-diags.txt)

edit: omitted `sudo apt update` in the above, attached file is correct.

**Comment by rickysarraf** [comment, marked as answer]
Please share full console logs. This isn't much useful

s3nt fr0m a $martph0ne, excuse typ0s

**Reply by buxbaumbixbybrae** [reply]
Please advise if something is missing from the original attached file, I had thought that was complete.

**Reply by rickysarraf** [reply]
Please follow this howto: https://github.com/rickysarraf/apt-offline/wiki

There's also a video demo linked in the howto page. That should guide you

**Reply by buxbaumbixbybrae** [reply]
Hi Thanks. I have read the wiki and the man page and also watched the video before posting.
There is a mistake in my post above where `sudo apt update` was omitted (now corrected), but the correct information is in the file attached to my first post `apt-offline-diags.txt`. 
Per the wiki and particularly the manpage I understand the basic commands to be:
```
offline$ sudo apt-offline set update.sig --update
online$  sudo apt-offline get update.sig --bundle update-bundle.zip
offline$ sudo apt-offline install update-bundle.zip
offline$ sudo apt update
offline$ sudo apt-offline set upgrade.sig --upgrade
--- completed to here. upgrade.sig is zero bytes ---
online$  sudo apt-offline get upgrade.sig --bundle upgrade-bundle.zip
offline$ sudo apt-offline install upgrade-bundle.zip
offline$ sudo apt upgrade
```
If that's wrong could you please highlight the error. I have done these in order but the file `upgrade.sig` is zero bytes and I am not sure what is my error or how to resolve / progress from here.

There are plenty of errors in the file`apt-offline-diags.txt` but I am not sure which ones are important and what to do about them, as some things have obviously worked. I would very much appreciate guidance on this.

To further clarify there are no errors generated by either of the `apt-offline set` commands.

The [manpage](http://manpages.ubuntu.com/manpages/focal/man8/apt-offline.8.html) under Examples `NOTE2:  On  a  fresh  setup installed through CD/DVD [...] add the default APT network repositories to the list`, does this apply in this case? And if so what should be done on a ubuntu box?


Thanks in advance and Regards

**Reply by rickysarraf** [reply]
```
offline$ sudo apt-offline set update.sig --update
online$  sudo apt-offline get update.sig --bundle update-bundle.zip
offline$ sudo apt-offline install update-bundle.zip
offline$ sudo apt update
offline$ sudo apt-offline set upgrade.sig --upgrade
--- completed to here. upgrade.sig is zero bytes ---
online$  sudo apt-offline get upgrade.sig --bundle upgrade-bundle.zip
offline$ sudo apt-offline install upgrade-bundle.zip
offline$ sudo apt upgrade
```

You don't need this step: `offline$ sudo apt update`

**Reply by rickysarraf** [reply]
Also, note that the version of `apt-offline` you are using is quite old. If possible, please try the latest `1.8.4` version, which is available in Ubuntu Jammy.

https://packages.ubuntu.com/jammy/apt-offline

**Reply by buxbaumbixbybrae** [reply]
Thank you very much, everything works fine under 1.8.4.
We have successfully performed update and upgrade steps and then installed new packages using apt-offline.
We initially used 1.8.2 because it was listed under 20.04 which is what our offline machine is running.
Your comment that `sudo apt update` is not required, is this always the case (I think so)?

Thanks again and Regards

**Reply by rickysarraf** [reply]
Yes. You are not supposed to run the `sudo apt update` command on the `offline` machine. That will not work because your offline machine has no internet connectivity. :-)

I'm glad to know that you finally got it all working and found `apt-offline` to be useful.
